JustSaying.Sample.Restaurant.Models/OrderPlacedEvent.cs

```diff
@@ -0,0 +1,11 @@
+﻿using JustSaying.Models;
+
+namespace JustSaying.Sample.Restaurant.Models
+{
+    public class OrderPlacedEvent : Message
```

Comment:
Simple POCOs with public gets and sets and no constructor are fine.

Reply:
Agreed, a private constructor could trip up the deserialization

Reply:
Isn't an immutable type most appropriate so that the subscriber is unable to change any component of the message that was received (intentionally or otherwise).
Also the publisher is only able to construct messages in a valid fashion via the constructor.
(The serializer is fine with this type)
```cs
    public class OrderPlacedEvent : Message
    {
        public OrderPlacedEvent(int orderId, string description)
        {
            if (orderId <= 0)
            {
                throw new ArgumentException("orderId must be positive", nameof(orderId));
            }
            if (string.IsNullOrWhiteSpace(description))
            {
                throw new ArgumentException("description must be supplied", nameof(description));
            }

            OrderId = orderId;
            Description = description;
        }

        public int OrderId { get; private set; }

        public string Description { get; private set; }
    }
```

Reply:
We've had issues in the past where trying to force business rules (what's required, what's optional etc.) into the POCOs, which are ultimately for serialization in the world of JustSaying, have caused problems as applications have evolved over time.

The biggest one being when you need to add a new optional field, so you add an extra constructor overload that sets it so it's not a binary breaking change, but forget to add appropriate hints to the serializer. Then you get runtime errors because the serializer doesn't know which constructor to use.

It also makes it more verbose when you're writing tests.

Ideally in a production application, validation etc. is handled elsewhere, not in the internals of JustSaying as a consequence of message publishing of deserialization.

Reply:
Ah, understood - Thanks!